src/Contract/UnbalancedTx.purs

```diff
@@ -35,24 +65,166 @@ import Ctl.Internal.ProcessConstraints.Error
+import Data.Log.Tag as TagSet
+import Data.Maybe (Maybe(Nothing, Just))
+import Data.Newtype (unwrap)
+import Effect.Exception (error)
 
 -- | Create an `UnbalancedTx` given `ScriptLookups` and
 -- | `TxConstraints`. This should be called in conjuction with
 -- | `balanceTx` and  `signTransaction`.
-mkUnbalancedTx
+-- |
+-- | This is a 'non-throwing' variant; if you need the 'throwing' variant, use
+-- | `mkUnbalancedTx` instead.
+mkUnbalancedTxE
   :: ScriptLookups
   -> TxConstraints
   -> Contract (Either MkUnbalancedTxError UnbalancedTx)
-mkUnbalancedTx = PC.mkUnbalancedTxImpl
+mkUnbalancedTxE = PC.mkUnbalancedTxImpl
+
+-- | As `mkUnbalancedTxE`, but 'throwing'.
+mkUnbalancedTx :: ScriptLookups -> TxConstraints -> Contract UnbalancedTx
+mkUnbalancedTx lookups constraints =
+  mkUnbalancedTxE lookups constraints >>= case _ of
+    Left err -> throwError $ error $ explainMkUnbalancedTxError err
+    Right res -> pure res
+
+-- | Helper to pretty-print `MkUnbalancedTxError`s.
+explainMkUnbalancedTxError :: MkUnbalancedTxError -> String
+explainMkUnbalancedTxError = case _ of
+  CannotFindDatum -> "Cannot find datum"
+  CannotQueryDatum dh ->
+    "Querying for datum by datum hash ("
+      <> show dh
+      <>
+        ") failed: no datum found"
+  CannotConvertPOSIXTimeRange tr ttsErr ->
+    "Cannot convert POSIX time range to slot time range.\nRange: "
+      <> show tr
+      <> "\nReason: "
+      <>
+        explainPosixTimeToSlotError ttsErr
+  CannotSolveTimeConstraints tr tr' ->
+    "Unsolvable time constraints: " <> show tr <> " and " <> show tr' <>
+      " do not overlap."
+  CannotGetMintingPolicyScriptIndex ->
+    "Cannot get minting policy script index. This should be impossible.\n"
+      <> "Please report this as a bug here: "
+      <> bugTrackerLink
+  CannotGetValidatorHashFromAddress addr ->
+    "Cannot get a payment validator hash from address " <> show addr
+  CannotMakeValue _ tn _ ->
+    "Attempted to make an amount with the ADA currency symbol, and non-empty token name "
+      <> show tn
+      <>
+        ". This is not allowed, as the ADA currency symbol can only be combined with the empty token name."
+  CannotWithdrawRewardsPubKey spkh ->
+    "Cannot withdraw rewards, as pubkey " <> show spkh <> " is not registered"
+  CannotWithdrawRewardsPlutusScript pssv ->
+    "Cannot withdraw rewards from Plutus staking script " <> show pssv
+  CannotWithdrawRewardsNativeScript nssv ->
+    "Cannot withdraw rewards from native staking script " <> show nssv
+  DatumNotFound hash -> "Datum with hash " <> show hash <> " not found."
+  DatumWrongHash hash datum -> "Datum " <> show datum
+    <> " does not have the hash "
+    <> show hash
+  MintingPolicyHashNotCurrencySymbol mph ->
+    "Minting policy hash " <> show mph <>
+      " is not a CurrencySymbol. Please check the validity of the byte representation."
+  MintingPolicyNotFound mp -> "Minting policy "
+    <> show mp
```

Comment:
Please, make sure that this pretty-printer outputs hex strings for types that have byte representations (token names should be human-readable most of the time, see comment above). Do not use `show`.